### ISSUE TITLE
Early stopping fails to terminate training in non-distributed mode

### ISSUE THREAD

**Issue by matteotolloso** [ACEsuit/mace#1384]
### Internal Logic Error: Early stopping fails to terminate training in non-distributed mode in `mace/tools/train.py`

**Description**
In the training script located at `mace/tools/train.py`, the early stopping mechanism does not function correctly when running in non-distributed mode (`distributed=False`). While the logger correctly identifies that the patience limit has been reached and prints the "Stopping optimization" message, the training loop continues to execute until it reaches `max_num_epochs`.

**Steps to Reproduce**
1. Run a training session with `distributed=False`.
2. Set a small `patience` value and a large `max_num_epochs`.
3. Provide a validation set where the loss stops improving.
4. Observe the logs after the patience threshold is hit.

**Observed Behavior**
The log prints:
`INFO: Stopping optimization after X epochs without improvement`
However, the script immediately proceeds to the next epoch and continues training until `max_num_epochs` is reached, ignoring the termination request.

**Root Cause Analysis**
The issue stems from how the `exit_now` signal is managed within `mace/tools/train.py`:

1. **Initialization:** At the beginning of the `train()` function, `exit_now` is initialized as `None` if distributed training is disabled:
   ```python
   exit_now = torch.zeros(1, device=device) if distributed else None
   ```

2. **Signal Trigger:** When the `patience_counter` exceeds `patience`, the logic to trigger a stop is guarded by a check for `exit_now`:
   ```python
   if patience_counter >= patience:
       logging.info(f"Stopping optimization after {patience_counter} epochs...")
       if exit_now is not None:
           exit_now.fill_(1)
   ```
   In a standard local run, `exit_now` is `None`, so the signal is never updated.

3. **Loop Termination:** The actual `break` statement that exits the `while` loop is also wrapped in a conditional check for `exit_now`:
   ```python
   if exit_now is not None:
       torch.distributed.broadcast(exit_now, src=0)
       if exit_now == 1:
           break
   ```
   Because `exit_now` remains `None` throughout the local execution, the `break` statement is unreachable, and the loop continues.

**Expected Behavior**
The training loop should terminate immediately after the "Stopping optimization" message is logged, regardless of whether the training is distributed or running on a single device.

**Comment by nguyenhungvju** [comment]
Thank you very much for your sharing issuse solving. I has the same problem but for the stage two (after start swa). The stage one still early stopped at the patience choosing but the stage still run until max number of epoch. Is it the same problem with you?
